backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java

```diff
@@ -30,142 +32,122 @@
  * @author cedric.bodet@tngtech.com
  * @author Johannes.Najjar@tngtech.com
  */
-@View(name = "all", map = "function(doc) { if (doc.type == 'component') emit(null, doc._id) }")
-public class ComponentRepository extends SummaryAwareRepository<Component> {
-
-    private static final String RECENT_VIEW = "function(doc) { if(doc.type == 'component') { emit(doc.createdOn, doc._id) } }";
-
-    private static final String SUBSCRIBERS_VIEW =
-            "function(doc) {" +
+@Views({
+        @View(name = "all",
+                map = "function(doc) { if (doc.type == 'component') emit(null, doc) }"),
```

Comment:
Can you please explain the rationale for when you emit `doc._id` and when the complete `doc`?

Reply:
The goal was to migrate from `doc._id` to `doc` in as many places as possible. That is clearly not possible/good for functions which return a list of IDs. This PR might not be complete, since there are still possibilities for optimizations.